### PR TITLE
chore(gatsby): Migrate redux/actions/types and  redux/actions/restricted to TypeScript

### PR DESCRIPTION
**Pull request by 1natsu172**
## Description
Part of #21995 JS to TS project.

Migrate targets
* `redux/actions/types.js`
* `redux/actions/restricted.js`

## Related Issues
Related to #21995

### CODE REVIEW

File: packages/gatsby/src/redux/types.ts [hunk 1]
```diff
@@ -136,21 +146,64 @@ export interface IQueryExtractionBabelErrorAction {
...
+  plugin: IGatsbyPlugin
+  traceId?: string
+  payload: GraphQLSchema
+}
+
+export interface ICreateTypes {
+  type: `CREATE_TYPES`
+  plugin: IGatsbyPlugin
+  traceId?: string
+  payload: any
```
Comment (blainekasten) [inline packages/gatsby/src/redux/types.ts]: let's try to figure out what this is

File: packages/gatsby/src/redux/types.ts [hunk 2]
```diff
@@ -136,21 +146,64 @@ export interface IQueryExtractionBabelErrorAction {
...
+    exclude?: { types?: Array<string>; plugins?: Array<string> }
+    withFieldTypes?: boolean
+  }
+}
+
+export interface ICreateResolverContext {
+  type: `CREATE_RESOLVER_CONTEXT`
+  plugin: IGatsbyPlugin
+  traceId?: string
+  payload: object
```
Comment (blainekasten) [inline packages/gatsby/src/redux/types.ts]: let's try to figure out what this is

File: packages/gatsby/src/redux/actions/restricted.ts [hunk 3]
```diff
@@ -1,11 +1,36 @@
-// @flow
 const { camelCase } = require(`lodash`)
 const report = require(`gatsby-cli/lib/reporter`)
 const { parseTypeDef } = require(`../../schema/types/type-defs`)
```
Comment (blainekasten) [inline packages/gatsby/src/redux/actions/restricted.ts]: let's make these all import statements

File: packages/gatsby/src/redux/types.ts (hunk 1 above)
Comment (1natsu172) [reply packages/gatsby/src/redux/types.ts]: I tried to improve this.

File: packages/gatsby/src/redux/types.ts (hunk 2 above)
Comment (1natsu172) [reply packages/gatsby/src/redux/types.ts]: This too tried to improve.

File: packages/gatsby/src/redux/actions/restricted.ts (hunk 3 above)
Comment (1natsu172) [reply packages/gatsby/src/redux/actions/restricted.ts]: Changed to import statement.

**Comment by vladar** [comment]
Published in `gatsby@2.20.16`

**Comment by 1natsu172** [comment]
Thanks for the review and merge!

**Pull request merged**